### PR TITLE
Support Lerna Independent mode

### LINKED ISSUE #370: `auto canary` doesn't seem to work with lerna independent mode
**Describe the bug**

Auto throws in Lerna independent mode:

```
❯ yarn auto canary --pr 137
yarn run v1.13.0
$ /Users/egoist/dev/saber/node_modules/.bin/auto canary --pr 137
TypeError: Invalid Version: vindependent
    at new SemVer (/Users/egoist/dev/saber/node_modules/auto/node_modules/semver/semver.js:323:11)
    at compare (/Users/egoist/dev/saber/node_modules/auto/node_modules/semver/semver.js:614:10)
    at Function.gt (/Users/egoist/dev/saber/node_modules/auto/node_modules/semver/semver.js:643:10)
    at /Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:121:52
    at step (/Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:32:23)
    at Object.next (/Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:13:53)
    at fulfilled (/Users/egoist/dev/saber/node_modules/auto/dist/plugins/npm/index.js:4:58)
    at processTicksAndRejections (internal/process/task_queues.js:88:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**To Reproduce**

1. In a repo using Lerna independent mode
2. Run `auto canary` against a PR

**Expected behavior**

`auto canary` should publish a pre release

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. iOS] macOS

**Additional context**

Auto version: v4.15.5

### PR DESCRIPTION
**Pull request by hipstersmoothie**
# What Changed

Made lots of changes to support lerna independent mode. 

# Why

closes #370 

Todo:

- [x] Add tests
- [x] Add docs
- [ ] Add yourself to contributors (run `yarn contributors:add`)

### CODE REVIEW

**Comment by adierkens** [comment]
Published PR with canary version: `5.1.0-canary.373.4594`

**Comment by adierkens** [comment]
Published PR with canary version: `5.1.0-canary.373.4599`

File: src/plugins/npm/__tests__/npm.test.ts [hunk 1]
```diff
@@ -533,8 +546,27 @@ describe('canary', () => {
       }
     `;
 
-    const canaryVersion = '1.2.3-canary.123.1';
-    await hooks.canary.promise(canaryVersion);
+    await hooks.canary.promise(SEMVER.patch, '');
     expect(exec.mock.calls[0][1]).toContain('lerna');
   });
+
+  // test('works when PR has "skip-release" label', async () => {
```
Comment (adierkens) [inline src/plugins/npm/__tests__/npm.test.ts]: How well does independent mode work with skipped releases?

**Comment by adierkens** [comment]
Published PR with canary version: `5.1.0-canary.373.4628`

**Comment by adierkens** [comment]
Published PR with canary version: `5.1.0-canary.373.4633`

**Comment by adierkens** [comment]
Published PR with canary version: `5.1.0-canary.373.4638`

File: src/plugins/npm/__tests__/npm.test.ts (hunk 1 above)
Comment (hipstersmoothie) [reply src/plugins/npm/__tests__/npm.test.ts]: Independent mode doesn't really effect `auto version` much so the skip release label still works fine. this test was in the wrong place. Moved to auto.test.ts "canary"

**Comment by adierkens** [comment]
:rocket: PR was released in v6.0.0 :rocket:

**Pull request merged**